code/modules/chemistry/Chemistry-Tools.dm

```diff
@@ -780,27 +781,42 @@ proc/ui_describe_reagents(atom/A)
+	proc/remove_container(var/obj/container)
+		src.UpdateOverlays(null, "tube\ref[container]")
+		UnregisterSignal(container, COMSIG_ATTACKHAND)
+		UnregisterSignal(container, XSIG_OUTERMOST_MOVABLE_CHANGED)
+		UnregisterSignal(container, COMSIG_MOVABLE_MOVED)
+		src.connected_containers.Remove(container)
+
+	proc/remove_all_containers()
+		for(var/obj/container in src.connected_containers)
+			remove_container(container)
 
 	proc/try_adding_reagents_to_container(reagent, amount, sdata, temp_new, donotreact, donotupdate) //called when a reaction occurs inside the condenser flagged with "chemical_reaction = TRUE"
-		if(!current_container) //if we have no beaker, dump the reagents into condenser
+		if(src.connected_containers.len <= 0) //if we have no beaker, dump the reagents into condenser
 			src.reagents.add_reagent(reagent, amount, sdata, temp_new, donotreact, donotupdate)
 		else
-			var/remaining_container_space = current_container.reagents.maximum_volume - current_container.reagents.total_volume
-			if(remaining_container_space < amount) 																			//if there's more reagent to add than the beaker can hold...
-				current_container.reagents.add_reagent(reagent, remaining_container_space, sdata, temp_new, donotreact, donotupdate) //...add what we can to the beaker...
-				src.reagents.add_reagent(reagent, amount - remaining_container_space, sdata, temp_new, donotreact, donotupdate)  //...then backflow remaining chems into the condenser
-
-			else
-				current_container.reagents.add_reagent(reagent, amount, sdata, temp_new, donotreact, donotupdate)
+			add_reagents_to_containers(reagent, amount, sdata, temp_new, donotreact, donotupdate)
+
+	proc/add_reagents_to_containers(reagent, amount, sdata, temp_new, donotreact, donotupdate)
+		var/list/non_full_containers = list()
+		for(var/obj/container in connected_containers)
+			if(container.reagents.maximum_volume > container.reagents.total_volume) //don't bother with this if it's already full, move onto other containers
+				non_full_containers.Add(container)
+		if(non_full_containers.len <= 0)	//all full? backflow!!
+			src.reagents.add_reagent(reagent, amount, sdata, temp_new, donotreact, donotupdate)
+		else
+			var/divided_amount = (amount / non_full_containers.len) //cut the reagents needed into chunks
+			for(var/obj/container in non_full_containers)
+				var/remaining_container_space = container.reagents.maximum_volume - container.reagents.total_volume
+				if(remaining_container_space < divided_amount) 																			//if there's more reagent to add than the beaker can hold...
+					container.reagents.add_reagent(reagent, remaining_container_space, sdata, temp_new, donotreact, donotupdate) //...add what we can to the beaker...
+					src.add_reagents_to_containers(reagent, divided_amount - remaining_container_space, sdata, temp_new, donotreact, donotupdate)  //...then run the whole proc again with the remaining reagent, evenly distributing to remaining containers
```

Comment:
should probably return after the recursion, otherwise it's going to iterate over all of the remaining containers again?

Reply:
The recursion here is specifically running the remainder of a beaker that is taking in more than it can hold through the other remaining beakers, if I make it return here beakers that are checked after won't get the initial bit they need. IE: If you mix 30u into three beakers, each one will get 10u, but if one's only able to take 5, the other two will split that 2.5 between themselves, but returning means they wouldn't get the initial 10u if they were later in the list, or that's what I've seen in testing.